### PR TITLE
Added the Ion Rifle

### PR DESCRIPTION
**Pull request by Boaz1111**
## About the PR

Added the ion rifle, as well as an ion rifle crate. In the future, one rifle should be mapped in every armory.
sprites as added in this PR https://github.com/space-wizards/space-station-14/pull/20949
## Why / Balance

Counter to borgs, laser rifles, other things using batteries. Mechs will be affected in the future as soon as they use standardised powercells just like borgs. 
## Media

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl:
- add: An Ion Rifle has been added to the game, it can be purchased by cargo, or found in the armory(soon)

### CODE REVIEW

**Comment by Boaz1111** [comment]
sorry for the laser rifle change, noticed a single sling pixel was not colored properly

File: Resources/Prototypes/Entities/Objects/Weapons/Guns/Projectiles/projectiles.yml [hunk 1]
```diff
@@ -949,3 +949,48 @@
...
+    damage:
+      types:
+        Shock: 5
+    soundHit:
+      collection: WeakHit
+    forceSound: true
+  - type: EmpOnTrigger
+    range: 2
+    energyConsumption: 200
+    disableDuration: 5
```
Comment (metalgearsloth) [inline Resources/Prototypes/Entities/Objects/Weapons/Guns/Projectiles/projectiles.yml:996]: This shouldn't be relying on triggersystem, it should just be its own component for EMPs.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

File: Resources/Prototypes/Entities/Objects/Weapons/Guns/Projectiles/projectiles.yml (hunk 1 above)
Comment (Boaz1111) [reply Resources/Prototypes/Entities/Objects/Weapons/Guns/Projectiles/projectiles.yml:996]: What do you mean?

Comment (metalgearsloth) [reply Resources/Prototypes/Entities/Objects/Weapons/Guns/Projectiles/projectiles.yml:996]: This has nothing to do with voice triggers or grenades which is what trigger is for.

Comment (deltanedas) [reply Resources/Prototypes/Entities/Objects/Weapons/Guns/Projectiles/projectiles.yml:996]: then it has copy paste EmpOnSpawn bloat code

Comment (Boaz1111) [reply Resources/Prototypes/Entities/Objects/Weapons/Guns/Projectiles/projectiles.yml:996]: Sloth RPG's, EMP grenades shot from ship guns, every explosive projectile, all do this

Comment (metalgearsloth) [reply Resources/Prototypes/Entities/Objects/Weapons/Guns/Projectiles/projectiles.yml:996]: Grenades are different, I'll just fix it.

**Comment by Ratyyy** [comment]
I think it's shit code pretty much, but here's the code for this pr:
``SubscribeLocalEvent<EmpOnCollideComponent, ProjectileHitEvent>(OnProjectileHit);``(to server EMP system, ~29 line)
``   private void OnProjectileHit(EntityUid uid, EmpOnCollideComponent component, ref ProjectileHitEvent args)
    {
        TryEmpEffects(args.Target, component.EnergyConsumption, component.DisableDuration);
    }``(to server EMP system, ~148-151 line)

and here's component:

`namespace Content.Server.Emp;

[RegisterComponent]
[Access(typeof(EmpSystem))]
public sealed partial class EmpOnCollideComponent : Component
{

    /// <summary>
    /// How much energy will be consumed per battery
    /// </summary>
    [DataField("energyConsumption"), ViewVariables(VVAccess.ReadWrite)]
    public float EnergyConsumption;

    /// <summary>
    /// How long it disables targets in seconds
    /// </summary>
    [DataField("disableDuration"), ViewVariables(VVAccess.ReadWrite)]
    public float DisableDuration = 60f;
}
`

**Comment by Boaz1111** [comment]
@metalgearsloth is that good enough code?

**Comment by Emisse** [comment]
Closed due to feature freeze May 10th-June 14th. Comment to have it reopen after this.

**Comment by Boaz1111** [comment]
Open

File: Resources/Prototypes/Entities/Objects/Weapons/Guns/Projectiles/projectiles.yml [hunk 2]
```diff
@@ -949,3 +949,48 @@
...
+  - type: Fixtures
+    fixtures:
+      projectile:
+        shape:
+          !type:PhysShapeAabb
+          bounds: "-0.15,-0.3,0.15,0.3"
+        hard: false
+        mask:
+        - Impassable
+        - BulletImpassable
```
Comment (metalgearsloth) [inline Resources/Prototypes/Entities/Objects/Weapons/Guns/Projectiles/projectiles.yml:982]: Any strong reason why it needs custom bounds?

File: Resources/Prototypes/Entities/Objects/Weapons/Guns/Projectiles/projectiles.yml [hunk 3]
```diff
@@ -949,3 +949,48 @@
...
     tags:
       - HideContextMenu
+
+- type: entity
+  name : ion rifle bolt
+  id: BulletIonRifle
+  parent: BaseBulletTrigger
+  noSpawn: true
+  components:
+  - type: Reflective
```
[the remark this answers is not in the document]
Comment (Boaz1111) [reply Resources/Prototypes/Entities/Objects/Weapons/Guns/Projectiles/projectiles.yml:959]: Yeah, shit, I know, it's also like currently not 100% functional

File: Resources/Prototypes/Entities/Objects/Weapons/Guns/Projectiles/projectiles.yml (hunk 2 above)
Comment (Boaz1111) [reply Resources/Prototypes/Entities/Objects/Weapons/Guns/Projectiles/projectiles.yml:982]: nope, my bad

**Comment by Emisse** [comment]
are you still working on this

**Comment by Boaz1111** [comment]
No I stopped, got too busy with other shit and this became frustrating

**Comment by slarticodefast** [comment]
> No I stopped, got too busy with other shit and this became frustrating

I'll close this PR then for now. If you want to continue working on it, just ping me and I will re-open it.

**Pull request closed without merging**